App/Controllers/LoginController.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+
+namespace App\Controllers;
+
+use Core\View;
+use App\Models\User;
+use App\Models\Permission;
+
+class LoginController {
+
+    public function index()
+    {
+        $users = User::getDB()->readAll(User::getTableName(),User::class);
+        $userExists = false;
+        if (count($users) > 0)
+        {
+            $userExists = true;
+        }
+        View::render("login.view.php",["userExists" => $userExists]);
+    }
+
+    public function login()
+    {
+      if(session_status() !== PHP_SESSION_ACTIVE)
+          session_start();
+
+      if(count($_POST)>0)
+      {
+         $sqlParams['user_name']=$_POST["user_name"];
+         $sqlParams['password']=$_POST["password"];
+
+         $users = User::getDB()->read(User::getTableName(),$sqlParams,User::class);
+
+         if (count($users) == 1)
+         {
+             $user = $users[0];
+
+             $sqlParams= [];
+             $sqlParams['role_id']=$user->getRoleId();
+             $permissions = Permission::getDB()->read(Permission::getTableName(),$sqlParams,Permission::class);
```

Comment:
Will not add on all models this for focusing also on other type of comments. Create a Repository for each Model and replace the logics you have around.